lib/RecommendationProvider.js

```diff
@@ -0,0 +1,157 @@
+      this._currentRecommendation = Object.assign({}, {
+        url: allowedRecommendations[index].url,
+        timestamp: allowedRecommendations[index].timestamp,
+        recommended: true});
+      return this._currentRecommendation;
+    }
+    this._currentRecommendation = null;
+    return this._currentRecommendation;
+  },
+
+  /**
+    * Fetches Pocket recommendations and saves them to a global list of recommended content
+    */
+  asyncSetRecommendedContent: Task.async(function*() {
+    try {
+      let response = yield this._asyncGetRecommendationsFromPocket();
+      if (response.ok) {
+        let responseJson = yield response.json();
+        if (responseJson.urls.length > 0) {
+          // wait for the recommendation's metadata to come back (we don't want to show ugly recommendations)
+          let richRecommendations = yield this._previewProvider.getLinkMetadata(responseJson.urls);
+          this._recommendedContent = richRecommendations;
+          this._recommendedContent.forEach(item => item.timestamp = this._convertTime(item.timestamp));
+        }
+      }
+    } catch (err) {
+      Cu.reportError(err);
+      throw err;
+    }
+  }),
+
+  /**
+    * Update the list of recommendations the user has blocked (they do not want to see these again)
+    */
+  setBlockedRecommendation(link) {
+    this._blockedRecommendedContent.push(link);
+    this._currentRecommendation = null;
+  },
+
+  /**
+    * Convert milliseconds into human readable form and style it accordingly
+    */
+  _convertTime(timestamp) {
+    let publishedTime = Date.now() - timestamp;
+    if (timestamp === 0 || publishedTime < MILLISECONDS_IN_TEN_MINUTES) {
+      return "Now";
+    } else if (publishedTime >= MILLISECONDS_IN_TEN_MINUTES && publishedTime < MILLISECONDS_IN_AN_HOUR) {
+      return Math.floor(((publishedTime / (MILLISECONDS_IN_A_MINUTE)) % 60)) + "m";
+    } else if (publishedTime >= MILLISECONDS_IN_AN_HOUR && publishedTime < MILLISECONDS_IN_A_DAY) {
+      return Math.floor(((publishedTime / (MILLISECONDS_IN_AN_HOUR)) % 24)) + "h";
+    } else if (publishedTime >= MILLISECONDS_IN_A_DAY && publishedTime < MILLISECONDS_IN_A_WEEK) {
+      return Math.floor(((publishedTime / (MILLISECONDS_IN_A_DAY)) % 7)) + "d";
+    } else {
+      return Math.floor((publishedTime / (MILLISECONDS_IN_A_WEEK))) + "w";
+    }
+  },
+
+  _asyncGetRecommendationsFromPocket: Task.async(function*() {
+    try {
+      let response = yield fetch(this._pocketEndpoint, {
+        method: "GET",
+        headers: {"Content-Type": "application/json"}
+      });
+      return response;
+    } catch (err) {
+      Cu.reportError(err);
+      throw err;
+    }
+  }),
+
+  /**
+    * Update the list of recommendations from Pocket
+    */
+  _asyncExpireRecommendations: Task.async(function*(pocketTimeout) {
+    if (pocketTimeout) {
+      this._pocketTimeoutID = setTimeout(() => {
+        this.asyncSetRecommendedContent();
```

Comment:
I mean, can you have a default value passed in this function, e.g.

```javascript
_asyncExpireRecommendations: Task.async(function*(pocketTimeout = 5000) {
}
```

The reason is that this function seems to require a parameter to function at all and it might be developer friendly for this to work especially since there is no status returned.